### PR TITLE
mk edits

### PR DESCRIPTION
**Pull request by mjkozicki**

### CODE REVIEW

File: tutorials/catchpoint/index.md [hunk 1]
```diff
@@ -1,215 +1,216 @@
...
+# Rework
+- how to enable enhanced capabilities to leverage BigQuery for querying the large data set metrics
+- how to realize the vision of a single pane for all network performance monitoring and analysis
 
-## Catchpoint Test Types
+## Catchpoint Supported Test Types
 
-The different types of test that Catchpoint supports are:
+- Synthetic Monitoring from any location
+  - Create transactional tests to replicate [Real User Monitoring](https://en.wikipedia.org/wiki/Real_user_monitoring#:~:text=Real%20user%20monitoring%20(RUM)%20is,server%20or%20cloud%2Dbased%20application.&text=The%20data%20may%20also%20be,intended%20effect%20or%20cause%20errors).
```
Comment (codebutmonitor) [inline tutorials/catchpoint/index.md:16]: Create transactional tests to replicate user workflows. This augments the passive Real User Monitoring Data.

File: tutorials/catchpoint/index.md [hunk 2]
```diff
@@ -1,215 +1,216 @@
...
+  - Layer 3 (traceroute, ping), Layer 4 (TCP, UDP), DNS and BGP.
 
-- Synthetic Monitoring from any location : Create transactional user-tests to replicate active [RUM](<https://en.wikipedia.org/wiki/Real_user_monitoring#:~:text=Real%20user%20monitoring%20(RUM)%20is,server%20or%20cloud%2Dbased%20application.&text=The%20data%20may%20also%20be,intended%20effect%20or%20cause%20errors.>)
-- Network Layer Tests from any location: Layer 3(traceroute,ping), Layer 4(TCP,UDP), DNS and BGP
+For more information, please refer to the [Catchpoint Platform](https://www.catchpoint.com/platform) documentation.
 
-For more information, please refer [here](https://www.catchpoint.com/platform). Once the relevant tests have been created, this guide highlights two approaches to consume the Catchpoint test data within GCP:
+This guide highlights two approaches to consume the Catchpoint test data within GCP
+
+# Need to fix these links, share the files
```
Comment (codebutmonitor) [inline tutorials/catchpoint/index.md:24]: These are just pointing to the respective sections in this file. I dont think we can support this in GitHub

File: tutorials/catchpoint/index.md [hunk 3]
```diff
@@ -1,215 +1,216 @@
...
-- [Creating the Catchpoint Setup](https://docs.google.com/document/d/1KuIc-A45aFJ3eK-Rm46nyf_bsR_TvPD2MhXU9Qd_Z_U/edit?ts=600b384e#heading=h.xpxkt04y4bh4)
+The steps from Catchpoint to Grafana
 
-1. GCP pipeline setup
+1. Catchpoint sends data via a predefined webhook. The template is configured within the Catchpoint portal (see next section for more details)
+2. Webhook runs in GCP using App Engine. **THIS IS WEIRD and i don't know what it means**
+3. App Engine uses [Pub-Sub](https://cloud.google.com/pubsub) to propagate data to configured channels.
+4. A [Dataflow](https://cloud.google.com/dataflow/) job listens to the Pub/Sub channel and inserts the a BigQuery accessible dataset.
+5. Data is processed to match the Catchpoint Schema and sent to [BigQuery](https://cloud.google.com/bigquery).
+6. Grafana uses a BigQuery data source to visualize the data.
```
Comment (codebutmonitor) [inline tutorials/catchpoint/index.md:42]: Lets add - "Graphana has a BigQuery plugin".

File: tutorials/catchpoint/index.md [hunk 4]
```diff
@@ -1,215 +1,216 @@
...
-- [GCP Pipeline Setup](https://docs.google.com/document/d/1KuIc-A45aFJ3eK-Rm46nyf_bsR_TvPD2MhXU9Qd_Z_U/edit?ts=600b384e#heading=h.ytup525q5dqj)
-- [Creating the Catchpoint Setup](https://docs.google.com/document/d/1KuIc-A45aFJ3eK-Rm46nyf_bsR_TvPD2MhXU9Qd_Z_U/edit?ts=600b384e#heading=h.xpxkt04y4bh4)
+The steps from Catchpoint to Grafana
 
-1. GCP pipeline setup
+1. Catchpoint sends data via a predefined webhook. The template is configured within the Catchpoint portal (see next section for more details)
+2. Webhook runs in GCP using App Engine. **THIS IS WEIRD and i don't know what it means**
+3. App Engine uses [Pub-Sub](https://cloud.google.com/pubsub) to propagate data to configured channels.
+4. A [Dataflow](https://cloud.google.com/dataflow/) job listens to the Pub/Sub channel and inserts the a BigQuery accessible dataset.
+5. Data is processed to match the Catchpoint Schema and sent to [BigQuery](https://cloud.google.com/bigquery).
```
Comment (codebutmonitor) [inline tutorials/catchpoint/index.md:41]: I think they are trying to tell that the Catchpoint schema is passed along. Not that we process the data to fit the Catchpoint scheme.

File: tutorials/catchpoint/index.md [hunk 5]
```diff
@@ -1,215 +1,216 @@
...
-In this section we'll explain each configuration step in detail. We will also provide links for explanations on how to set up each component and those scripts that are used in the examples. The configuration details has two main tasks:
+![The end-to-end data ingestion pipeline within GCP. Catchpoint to App Engine to Cloud Pub/Sub to Cloud Dataflow to BigQuery to Grafana](integration-pipeline.png)
 
-- [GCP Pipeline Setup](https://docs.google.com/document/d/1KuIc-A45aFJ3eK-Rm46nyf_bsR_TvPD2MhXU9Qd_Z_U/edit?ts=600b384e#heading=h.ytup525q5dqj)
-- [Creating the Catchpoint Setup](https://docs.google.com/document/d/1KuIc-A45aFJ3eK-Rm46nyf_bsR_TvPD2MhXU9Qd_Z_U/edit?ts=600b384e#heading=h.xpxkt04y4bh4)
+The steps from Catchpoint to Grafana
 
-1. GCP pipeline setup
+1. Catchpoint sends data via a predefined webhook. The template is configured within the Catchpoint portal (see next section for more details)
+2. Webhook runs in GCP using App Engine. **THIS IS WEIRD and i don't know what it means**
```
Comment (codebutmonitor) [inline tutorials/catchpoint/index.md:38]: An HTTP webhook is set up in GCP using App Engine to ingest Catchpoint data. This is elaborated in the next bullet points

File: tutorials/catchpoint/index.md [hunk 6]
```diff
@@ -1,215 +1,216 @@
...
+6. Grafana uses a BigQuery data source to visualize the data.
 
-1. Creating pub-sub topics to stream data to an ingestion pipeline
+### Configuration Details
 
-Before we deploy our application in App Engine, we need to create a Pub/Sub topic to receive and distribute data. Pub-sub enables data to be streamed to the final data source. Pub topic will host the data subject(receive data) and the sub that subscribes to the pub topic will forward data to streaming service
+This section covers each configuration step with further explanations to set up each component and example scripts. The configuration details describes two main tasks
 
-(Reference to create a Topic: [https://cloud.google.com/pubsub/docs/quickstart-console#create_a_topic](https://cloud.google.com/pubsub/docs/quickstart-console#create_a_topic))
+# Need to fix these links, share the files
```
Comment (codebutmonitor) [inline tutorials/catchpoint/index.md:48]: This again is a link to the respective section within this file. The document was created using work which allowed this functionality.

File: tutorials/catchpoint/index.md [hunk 7]
```diff
@@ -1,215 +1,216 @@
...
+#### BigQuery dataset and data tables
 
-Now that we have our data flowing into Pub/Sub, we need to move that data to BigQuery, so that we can run some analytics on the data. We need to create the BigQuery table before we can create the pipeline.
+Once the Pub/Sub topics are setup and data is flowing, BigQuery needs the data to run analytics. Before the pipeline can be created, the BigQuery table needs to be setup.
 
-Please refer to this link on how to create a BigQuery table: [https://cloud.google.com/bigquery/docs/tables](https://cloud.google.com/bigquery/docs/tables)
+[How to create a BigQuery table](https://cloud.google.com/bigquery/docs/tables)
 
-Create a BigQuery dataset in your project:
+##### Create a BigQuery dataset
```
Comment (codebutmonitor) [inline tutorials/catchpoint/index.md:88]: This doesnt have to be a separate bullet points

File: tutorials/catchpoint/index.md [hunk 8]
```diff
@@ -1,215 +1,216 @@
...
 
 ![create-big-query](create-big-query.png)
```
Comment (codebutmonitor) [inline tutorials/catchpoint/index.md:90]: This image is not showing up in the rich text version

File: tutorials/catchpoint/index.md [hunk 9]
```diff
@@ -1,215 +1,216 @@
...
+  --project=<gcp-project-name> \
+  --subscription=projects/<gcp-project-name>/subscriptions/<pub-sub-subscription> \
+  --jobName=<existing-pipeline-job-name>
+  --update
+```
 
-A sample deployed job will be like this:
+###### A sample deployed job
 
 ![sample-deployed-job](sample-deployed-job.png)
```
Comment (codebutmonitor) [inline tutorials/catchpoint/index.md:146]: This image is not getting displayed in rich text

File: tutorials/catchpoint/index.md [hunk 10]
```diff
@@ -1,215 +1,216 @@
...
-Grafana is available in the GCP market place. You can deploy Grafana in your GCP project using [this](https://console.cloud.google.com/marketplace/details/click-to-deploy-images/grafana?filter=category%3Amonitoring&id=7f529a97-13af-4bcd-8050-3c5a63369d33&authuser) link.
+Grafana is available in the GCP market place. A guide to deploy Grafana in your GCP project is available [here](https://console.cloud.google.com/marketplace/details/click-to-deploy-images/grafana).
 
-Grafana offers a plugin for BigQuery as a data source. You can find step by step instructions [here](https://grafana.com/grafana/plugins/doitintl-bigquery-datasource).
+Grafana offers a plugin for BigQuery as a data source. Step by step instructions are available [here](https://grafana.com/grafana/plugins/doitintl-bigquery-datasource).
 
-Below are some examples of Grafana dashboard displays:
+###### Sample Grafana dashboards
 
 ![grafana-dashboard-1](grafana-dashboard-1.png)
```
Comment (codebutmonitor) [inline tutorials/catchpoint/index.md:156]: These three images are not showing up in rich text

File: tutorials/catchpoint/index.md [hunk 11]
```diff
@@ -1,215 +1,216 @@
...
 
-1. Catchpoint setup
-
-Now we have the GCP ingestion pipeline built, we need to make the CatchPoint tests to post data to the GCP webhook.
-
-Follow these steps to create a webhook on CatchPoint portal:
+#### Catchpoint setup
 
-- On the [Catchpoint dashboard](https://portal.catchpoint.com/), select the &quot;GEAR&quot; icon from the left menu bar.
+After the GCP ingestion pipeline is built, CatchPoint tests must be configured to post data to the GCP webhook.
```
Comment (codebutmonitor) [inline tutorials/catchpoint/index.md:164]: Catchpoint.. (not P)

File: tutorials/catchpoint/index.md [hunk 12]
```diff
@@ -225,80 +226,55 @@ Configuration Steps
...
+1. Enable Monitoring API  
+The Monitoring API must be enabled and have authorized users. Follow the steps in [Enabling the Monitoring API | Cloud Monitoring](https://cloud.google.com/monitoring/api/enable-api) to enable and authorize use of the Monitoring API v3. The Monitoring API can be enabled using either the Cloud SDK or the Cloud console. Both these approaches have been explained in the referenced guide.
 
-const pubsub = new PubSub();
+1. Enable Cloud Functions  
+Follow the steps in [Cloud Pub/Sub Tutorial | Cloud Functions Documentation](https://cloud.google.com/functions/docs/tutorials/pubsub) to enable the use of Cloud Functions and Cloud Pub/Sub APIs. This tutorial leverages Node.js. The referenced guide contains information to setup your development environment.
 
-const googleProjectId = process.env.GoogleProjectId;
-```
+##### GCP data pipeline setup
```
Comment (codebutmonitor) [inline tutorials/catchpoint/index.md:241]: This is the 4th step. We coudl either mark this as 4 or make the above 3 as sub topics as well

File: tutorials/catchpoint/index.md [hunk 13]
```diff
@@ -225,80 +226,55 @@ Configuration Steps
...
+    const { PubSub } = require('@google-cloud/pubsub');
+    const path = require('path')
+    const dotenv = require('dotenv')
+    dotenv.config({ path: path.join(\_\_dirname, '.env') });
+    const pubsub = new PubSub();
+    const googleProjectId = process.env.GoogleProjectId;
+    ```
 
-1. To set/change the project property in the core section, run:  
+1. Creating cloud functions to stream data to an ingestion pipeline
```
Comment (codebutmonitor) [inline tutorials/catchpoint/index.md:259]: This is the 2nd main heading. Creating cloud function to stream data..This has 3 sub bullets

File: tutorials/catchpoint/index.md [hunk 14]
```diff
@@ -225,80 +226,55 @@ Configuration Steps
...
+2. Open Google Cloud SDK Shell and navigate to the directory where the Node.js scripts were cloned
```
Comment (codebutmonitor) [inline tutorials/catchpoint/index.md:260]: This is the 1st sub bullet under Creating cloud function.

File: tutorials/catchpoint/index.md [hunk 15]
```diff
@@ -225,80 +226,55 @@ Configuration Steps
...
+    dotenv.config({ path: path.join(\_\_dirname, '.env') });
+    const pubsub = new PubSub();
+    const googleProjectId = process.env.GoogleProjectId;
+    ```
 
-1. To set/change the project property in the core section, run:  
+1. Creating cloud functions to stream data to an ingestion pipeline
+2. Open Google Cloud SDK Shell and navigate to the directory where the Node.js scripts were cloned  
+    `cd <path to cloned directory>`
+3. To set/change the project property in the core section, run:
```
Comment (codebutmonitor) [inline tutorials/catchpoint/index.md:262]: This is the 2nd sub bullet under Creating cloud function.

File: tutorials/catchpoint/index.md [hunk 16]
```diff
@@ -225,80 +226,55 @@ Configuration Steps
...
 
-1. To set/change the project property in the core section, run:  
+1. Creating cloud functions to stream data to an ingestion pipeline
+2. Open Google Cloud SDK Shell and navigate to the directory where the Node.js scripts were cloned  
+    `cd <path to cloned directory>`
+3. To set/change the project property in the core section, run:  
    `gcloud config set project my-project`
 
-1. Next, we will deploy two cloud functions to handle data injection from Catchpoint. We will be leveraging Pub/Sub, an asynchronous messaging service that decouples services that produce events from services that process events. Refer to [https://cloud.google.com/pubsub/docs/overview](https://cloud.google.com/pubsub/docs/overview) for understanding more about how these functions are useful.
+4. Next, we will deploy two cloud functions to handle data injection from Catchpoint. We will be leveraging Pub/Sub, an asynchronous messaging service that decouples services that produce events from services that process events. Refer to [https://cloud.google.com/pubsub/docs/overview](https://cloud.google.com/pubsub/docs/overview) for understanding more about how these functions are useful.
```
Comment (codebutmonitor) [inline tutorials/catchpoint/index.md:265]: This is the 3rd sub bullet under Creating cloud function.

File: tutorials/catchpoint/index.md [hunk 17]
```diff
@@ -225,80 +226,55 @@ Configuration Steps
...
 
-__***TIP: Using Catchpoint REST APIs***__
+###### TIP: Using Catchpoint REST APIs
 
 _In case you need to leverage the PULL model to integrate with Catchpoint, you can look at the example that leverages Catchpoint REST APIs in the Github repository_ [_here_](https://github.com/catchpoint/Integrations.GoogleCloudMonitoring)_. The catchpoint-rest-api.js has the code to leverage the Catchpoint performance API to pull data from Catchpoint. If you use this approach, you can skip the cloud function step up and go over to the Cloud Monitoring Set Up._
 
-1. HTTP trigger
-
-HTTP triggers are used to invoke cloud functions with a HTTP request. More on this - [https://cloud.google.com/functions/docs/calling/http](https://cloud.google.com/functions/docs/calling/http)
+1. HTTP triggers are used to invoke cloud functions with an HTTP request. Additional information is available [here](https://cloud.google.com/functions/docs/calling/http).
```
Comment (codebutmonitor) [inline tutorials/catchpoint/index.md:274]: This is 3 a. Comes under deployment of cloud functions to handle data injection from Catchpoint.

File: tutorials/catchpoint/index.md [hunk 18]
```diff
@@ -323,30 +299,29 @@ exports.catchpointPublish = async (req, res) => {
...
+        --trigger-http \
+        --runtime nodejs10 \
+        --trigger-http \
+        --allow-unauthenticated
+    ```
 
-Copy the URL after the deploy is successful. You will need this to set up the webhook in Catchpoint in Step 6.1.
+2. Copy the URL after the deploy is successful. This is needed to set up the Catchpoint Webhook in Step 6.1.
 
 ![sample-publish-function](sample-publish-function.png)
```
Comment (codebutmonitor) [inline tutorials/catchpoint/index.md:316]: Image not showing up in rich text

File: tutorials/catchpoint/index.md [hunk 19]
```diff
@@ -323,30 +299,29 @@ exports.catchpointPublish = async (req, res) => {
...
 
-1. Pub/Sub Trigger
+1. Pub/Sub Trigger
```
Comment (codebutmonitor) [inline tutorials/catchpoint/index.md:318]: This is the second bullet point after HTTP Trigger

File: tutorials/catchpoint/index.md [hunk 20]
```diff
@@ -359,34 +334,33 @@ exports.catchpointSubscribe = (message) => {
...
+    ```
 
 ![sample-subscribe-function](sample-subscribe-function.png)
 
-Note: You can either create the topic separately before deploying as below or alternatively specify it directly while deploying. If the topic is not created, the above command automatically creates a new topic.
-
-```sh
-gcloud pubsub topics create _ **MY_TOPIC** _
-```
+1. Set the TopicName environment variable using the same topic name you used to deploy the Subscribe function. Update the TopicName variable in the .env file at [https://github.com/catchpoint/Integrations.GoogleCloudMonitoring/blob/master/Stackdriver-Webhook/.env](https://github.com/catchpoint/Integrations.GoogleCloudMonitoring/blob/master/Stackdriver-Webhook/.env). Refer to [https://cloud.google.com/functions/docs/env-var](https://cloud.google.com/functions/docs/env-var) for information on how to set up environment variables.
```
Comment (codebutmonitor) [inline tutorials/catchpoint/index.md:351]: This is c... 
a. HTTP Trigger
b. Pub/Sub Trigger 
c.Setting topic name

File: tutorials/catchpoint/index.md [hunk 21]
```diff
@@ -359,34 +334,33 @@ exports.catchpointSubscribe = (message) => {
...
 
-```javascript
-const topicName = process.env.TopicName;
-```
+    ```javascript
+    const topicName = process.env.TopicName;
+    ```
 
-1. Cloud Monitoring set up
+##### Cloud Monitoring set up
```
Comment (codebutmonitor) [inline tutorials/catchpoint/index.md:363]: This is the 5th step. We could either mark this and 4 numerically or make the above 3 as sub topics as well

File: tutorials/catchpoint/index.md (hunk 8 above)
Comment (mjkozicki) [reply tutorials/catchpoint/index.md:85]: They are showing up here: https://github.com/mjkozicki/community/blob/ef54a8a2ae7870915b51c7aec1bfecee63e4c506/tutorials/catchpoint/index.md

**Pull request closed without merging**